spec/knapsack_pro/runners/queue/cucumber_runner_spec.rb

```diff
@@ -117,36 +117,50 @@
 
         expect(ENV).to receive(:[]=).with('KNAPSACK_PRO_BEFORE_QUEUE_HOOK_CALLED', 'true')
 
-        expect($?).to receive(:exitstatus).and_return(exitstatus)
+        expect($?).to receive(:exited?).and_return(process_exited)
+        allow($?).to receive(:exitstatus).and_return(exitstatus)
```

Comment:
I guess here we could have allow on `process_exited` to be more aligned with code where we call `process_status.exitstatus`

Reply:
Do you mean changing this to `allow(process_status).to receive(:exited?)`? If so, then I'm not sure I agree with that as that would mean the test knows the internals of the tested class too much in my opinion.

By mocking `$?`'s method, we essentially just mock the outside world and don't care how it's implemented in the class. In other words, I don't believe this test should start failing when we change the naming of the local variable in the class.

Reply:
right, I wrongly assumed we already have mocked $? here but it's not needed. No changes here.